### PR TITLE
clementine: use helpers, fix exception

### PR DESCRIPTION
**Pull request by lasers**
Uses `self.py3.command_output()`

The code will need to be worked on... to take care of exceptions when clementine isn't running. Also confirmed on `master`. I can fix it later. Just for this PR, I introduces helpers.

EDIT: Closes https://github.com/ultrabug/py3status/issues/713.

### CODE REVIEW

File: py3status/modules/clementine.py [hunk 1]
```diff
@@ -32,11 +30,9 @@ def _getMetadatas(self):
...
         Get the current song metadatas (artist - title)
         """
-        track_id = check_output(CMD + '.GetCurrentTrack', shell=True)
-        metadatas = check_output(
-            CMD + '.GetMetadata {}'.format(track_id.decode()), shell=True
-        )
-        lines = metadatas.decode('utf-8').split('\n')
+        track_id = self.py3.command_output(CMD + '.GetCurrentTrack')
+        metadatas = self.py3.command_output(CMD + '.GetMetadata {}'.format(track_id))
+        lines = metadatas.split('\n')
```
Comment (tobes) [inline py3status/modules/clementine.py]: `splitlines()` is better practice than `split('\n')`

Comment (lasers) [inline py3status/modules/clementine.py]: Pushed. Will record the exception.

**Comment by tobes** [comment]
@lasers can you just confirm you have tested this in python2/3.  It looks fine but I've not got the needed dependencies

**Comment by lasers** [comment]
Everything ran okay in P2/P3. Installed, not-installed, etc. I see one thing though.
With clementine running... 
```
^C%lasers~/src/py3status/py3status/modules git:(help-clementine) python2 clementine.py
None
None
None

C%lasers~/src/py3status/py3status/modules git:(help-clementine) python3 clementine.py 
{'cached_until': 1486645491.0, 'full_text': '♫ '}
{'cached_until': 1486645492.0, 'full_text': '♫ '}
{'cached_until': 1486645493.0, 'full_text': '♫ '}
```

**Comment by tobes** [comment]
ok so it's broken then.  what is lines looking like for both?

I think the whole `if lines:` line looks wrong but I'm not really sure what the module is wanting to do or why.  Maybe I look at the old code some point

**Comment by lasers** [comment]
Leaving for a blood work. Be back in an hour. Python2 OK in `master`.
```
lasers~/src/py3status/py3status/modules git:(master) python2 clementine.py
{'full_text': u'\u266b ', 'cached_until': 1486646082.0}
{'full_text': u'\u266b ', 'cached_until': 1486646083.0}
^C%`
```
No song to play, should print out `♫ ` Will compare my work when I get back.

**Comment by tobes** [comment]
> blood work. 

you vampire?

**Comment by lasers** [comment]
@tobes We don't need `if lines:` anymore because of try/except. New state pushed.
Everything tested OK now. *[slurping blood]* No... Not a vampire.

File: py3status/modules/clementine.py [hunk 2]
```diff
@@ -28,54 +30,55 @@ class Py3status:
...
+            lines = filter(None, metadata.splitlines())
+        except:
+            return {
+                'cached_until': self.py3.time_in(self.cache_timeout),
+                'color': self.py3.COLOR_BAD,
+                'full_text': STRING_ERROR
+            }
+
+        for item in lines:
+            if item.find('artist:') != -1:
```
Comment (tobes) [inline py3status/modules/clementine.py]: I don't really like all this `if item.find('artist:') != -1:` we could just do `if 'artist' in item:` to me this is more readable - you can keep the `find()` if you want.

**Comment by lasers** [comment]
@ultrabug This ready, sir.

**Comment by ultrabug** [comment]
Thanks a lot !

**Pull request merged**